ee/codegen/src/project.ts

```diff
@@ -362,13 +362,60 @@ ${errors.slice(0, 3).map((err) => {
       this.workflowContext.addInputVariableContext(inputVariableContext);
     });
 
-    this.workflowVersionExecConfig.outputVariables.forEach((outputVariable) => {
-      const outputVariableContext = new OutputVariableContext({
-        outputVariableData: outputVariable,
-        workflowContext: this.workflowContext,
+    // TODO: Invert / remove this logic once output values are default and terminal nodes don't exist.
+    // We are prioritizing terminal nodes as a temporary workaround to bad data from output values lingering in workflows.
+    const terminalNodes =
+      this.workflowVersionExecConfig.workflowRawData.nodes.filter(
+        (nodeData): nodeData is FinalOutputNodeType =>
+          nodeData.type === "TERMINAL"
+      );
+
+    if (terminalNodes.length > 0) {
+      // Create from terminal nodes
+      terminalNodes.forEach((nodeData) => {
+        const outputVariableContext = new OutputVariableContext({
+          outputVariableData: {
+            id: nodeData.data.outputId,
+            key: nodeData.data.name,
+            type: nodeData.data.outputType,
+          },
+          workflowContext: this.workflowContext,
+        });
+        this.workflowContext.addOutputVariableContext(outputVariableContext);
+
+        const workflowOutputContext = new WorkflowOutputContext({
```

Comment:
In previous PRs, I mentioned eventually deleting WorkflowOutputContext.

This PR shows that we now instantiate the two together, showing that we could in the future. But this gets me far enough that I need to solve the original output related issues that started this series of PRs